maildump/smtp.py

```diff
@@ -1,17 +1,115 @@
+import base64
 import smtpd
 from email.parser import BytesParser
 
 from logbook import Logger
+from passlib.apache import HtpasswdFile
 
 from maildump.db import add_message
 
 log = Logger(__name__)
 
 
+class SMTPChannel(smtpd.SMTPChannel, object):
+    def __init__(self, server, conn, addr, smtp_auth):
+        super(SMTPChannel, self).__init__(server, conn, addr)
+        self._smtp_auth = smtp_auth
+        self._authorized = False
+
+    def is_valid_user(self, auth_data):
+        auth_data_splitted = auth_data.split(b'\x00')
+        if len(auth_data_splitted) != 3:
+            return False
+
+        if not auth_data.startswith(b'\x00') and auth_data_splitted[0] != auth_data_splitted[1]:
+            return False
+
+        return self._smtp_auth.check_password(auth_data_splitted[1], auth_data_splitted[2])
+
+    def smtp_EHLO(self, arg):
```

Comment:
Do we really need to override all this? A quick google search pointed me to [this patch](https://bugs.python.org/file35991/smtpd_AUTH_full2.patch) from this [python issue](https://bugs.python.org/issue21935) where someone tried to get smtp auth support upstream, and they don't override EHLO. However, I don't know how good that implementation is.

Reply:
I don't know how in other way push info about AUTH PLAIN to capabilities.